### PR TITLE
[GSProcessing] Add support for EMR on EC2 execution.

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* Add Dockerfile for EMR on EC2.
* Add instructions in documentation.
* Modify Docker build and push scripts accordingly.

### Testing

Launched test clusters and jobs using EMR 7.0.0 and 6.10.0.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/gs-processing/usage/emr-serverless.rst [hunk 1]
```diff
@@ -69,7 +69,7 @@ you are using would be:
                     "ecr:BatchGetImage",
                     "ecr:DescribeImages"
                 ],
-                "Resource": "<enter-ecr-repository-arn-here>"
+                "Resource": ""<ACCOUNT>.dkr.ecr.<REGION>.amazonaws.com/graphstorm-processing-emr-serverless"
```
Comment (jalencato) [inline docs/source/gs-processing/usage/emr-serverless.rst:72]:
```suggestion
                "Resource": "<ACCOUNT>.dkr.ecr.<REGION>.amazonaws.com/graphstorm-processing-emr-serverless"
```

File: graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# TODO: Pin image version
+FROM public.ecr.aws/amazoncorretto/amazoncorretto:17 as base
```
Comment (jalencato) [inline graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu:2]: A general comment, is the difference between emr and emr-serverless only about basic image?

File: docs/source/gs-processing/usage/distributed-processing-setup.rst [hunk 3]
```diff
@@ -87,14 +87,14 @@ Once Docker and Poetry are installed, and your AWS credentials are set up,
 we can use the provided scripts
 in the ``graphstorm-processing/docker`` directory to build the image.
 
-GSProcessing supports Amazon SageMaker and EMR Serverless as
+GSProcessing supports Amazon SageMaker, EMR, and EMR Serverless as
```
Comment (jalencato) [inline docs/source/gs-processing/usage/distributed-processing-setup.rst:90]:
```suggestion
GSProcessing supports Amazon SageMaker, EMR on EC2, and EMR Serverless as
```

**Review by jalencato — APPROVED** [review]
Approve. Just some minor comment.

File: graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu (hunk 2 above)
Comment (thvasilo) [reply graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu:2]: No the EMR on EC2 image requires extra dependencies to be installed as it's more "bare bones" than the EMR-S image.

File: graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+# Install GSProcessing requirements to pyenv Python
+COPY requirements.txt requirements.txt
+# Use --mount=type=cache,target=/root/.cache when Buildkit CI issue is fixed:
+# https://github.com/moby/buildkit/issues/1512
+RUN pip3 install -r /usr/lib/spark/code/requirements.txt \
+    && rm -rf /root/.cache
+
+# Install Huggingface model cache if it is necessary
+# This needs to happen after the transformers library has been installed above
+ARG MODEL=""
```
Comment (classicsong) [inline graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu:57]: Do we have any doc about caching HF models in the docker image?

**Review by classicsong — APPROVED** [review]

File: graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu (hunk 4 above)
Comment (thvasilo) [reply graphstorm-processing/docker/0.3.0/emr/Dockerfile.cpu:57]: Yup it's here: https://graphstorm.readthedocs.io/en/latest/gs-processing/usage/distributed-processing-setup.html#packaging-huggingface-models-into-the-image

**Pull request merged**